server/langserver/langserver.go

```diff
@@ -0,0 +1,76 @@
+package langserver
+
+import (
+	"context"
+	"net"
+
+	"github.com/livegrep/livegrep/server/config"
+	"github.com/livegrep/livegrep/server/log"
+	"github.com/sourcegraph/jsonrpc2"
+	"path/filepath"
+	"time"
+)
+
+// infers a language server for a given file. Picks only one.
+func ForFile(repo *config.RepoConfig, filePath string) *config.LangServer {
+	fileExt := filepath.Ext(filePath)
+	for _, langServer := range repo.LangServers {
+		for _, ext := range langServer.Extensions {
+			if ext == fileExt {
+				return &langServer
+			}
+		}
+	}
+	return nil
+}
+
+type Client interface {
+	Initialize(ctx context.Context, params *InitializeParams) (InitializeResult, error)
+	JumpToDef(ctx context.Context, params *TextDocumentPositionParams) ([]Location, error)
+}
+
+type langServerClientImpl struct {
+	rpcClient *jsonrpc2.Conn
+}
+
+type handler struct{}
+
+func (h handler) Handle(context.Context, *jsonrpc2.Conn, *jsonrpc2.Request) {}
+
+func NewClient(ctx context.Context, address string) (client Client, err error) {
+	codec := jsonrpc2.VSCodeObjectCodec{}
+	conn, err := net.Dial("tcp", address)
+	if err != nil {
+		return
+	}
+	rpcConn := jsonrpc2.NewConn(ctx, jsonrpc2.NewBufferedStream(conn, codec), handler{})
+	client = &langServerClientImpl{
+		rpcClient: rpcConn,
+	}
+	return
+}
+
+func (ls *langServerClientImpl) Initialize(ctx context.Context, params *InitializeParams) (result InitializeResult, err error) {
+	err = ls.invoke(ctx, "initialize", params, &result)
+	if err != nil {
+		ls.invoke(ctx, "initialized", nil, nil)
```

Comment:
It seems that "initialized" is only invoked if there is an error on the call to initialize, also the error from invoking "initialized" isn't propagated back here if one were to occur

Reply:
Actually looked into this a little more, and I think you want to use 
`Notify` instead of `Call` when invoking `initialized` as it's a notification not an RPC call
https://sourcegraph.com/github.com/sourcegraph/jsonrpc2/-/blob/jsonrpc2.go#L455